src/Auth.ts

```diff
@@ -41,8 +47,35 @@ export enum CloudType {
   China
 }
 
-export class Service {
+export interface Identity {
+  connected: boolean;
+  identityName?: string;
+  identityId?: string;
+  authType: AuthType;
+  userName?: string;
+  password?: string;
+  secret?: string;
+  certificateType: CertificateType;
+  certificate?: string;
+  thumbprint?: string;
+  accessTokens: Hash<AccessToken>;
+  spoUrl?: string;
+  tenantId?: string;
+  // ID of the Azure AD app used to authenticate
+  appId: string;
+  // ID of the tenant where the Azure AD app is registered; common if multitenant
+  tenant: string;
+  cloudType: CloudType;
+}
+
+export class Service implements Identity {
```

Comment:
Let's rename tenantId to `spoTenantId` as it's different than the tenant ID that's on the access token and is specific for CSOM